examples/amm/src/contract.rs

```diff
@@ -312,17 +312,10 @@ impl AmmContract {
     /// authenticate the originator of the message
     fn check_account_authentication(&mut self, owner: AccountOwner) {
         match owner {
-            AccountOwner::User(address) => {
-                assert_eq!(
-                    self.runtime.authenticated_signer(),
-                    Some(address),
-                    "Unauthorized"
-                )
-            }
-            AccountOwner::Application(id) => {
-                assert_eq!(
-                    self.runtime.authenticated_caller_id(),
-                    Some(id),
+            AccountOwner::Address32(address) => {
+                assert!(
+                    self.runtime.authenticated_signer().map(|o| o.0) == Some(address)
```

Comment:
nit: we don't use single-letter variables as much, except `x`, `i`, `n`.

Reply:
Done.